.github/workflows/test.yml

```diff
@@ -1,52 +1,73 @@
-name: CI
-on:
-  pull_request:
-    paths-ignore:
-    - '**.md'
-    - '**.org'
-    - '.dir-locals.el'
-    - '.elpaignore'
-    - '.gitignore'
-    - 'doc/**'
-    branches:
-      - master
-  push:
-    paths-ignore:
-    - '**.md'
-    - '**.org'
-    - '.dir-locals.el'
-    - '.elpaignore'
-    - '.gitignore'
-    - 'doc/**'
-permissions: {}
-jobs:
-  build:
-    runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        emacs_version:
-          - 28.2
-          - 29.4
-          - 30.1
-          - snapshot
-
-    steps:
-    - uses: purcell/setup-emacs@master
-      with:
-        version: ${{ matrix.emacs_version }}
-
-    - uses: actions/checkout@v4
-      with:
-        persist-credentials: false
-
-    - name: Check Emacs version
-      run: emacs --version
-
-    - name: Build
-      run: |
-        make elpa
-        make build
-
-    - name: Test
-      run: make test-local
+name: CI
+
+on:
+  pull_request:
+    paths-ignore:
+    - '**.md'
+    - '**.org'
+    - '.dir-locals.el'
+    - '.elpaignore'
+    - '.gitignore'
+    - 'doc/**'
+    branches: [main, master]
+  push:
+    paths-ignore:
+    - '**.md'
+    - '**.org'
+    - '.dir-locals.el'
+    - '.elpaignore'
+    - '.gitignore'
+    - 'doc/**'
+    branches: [main, master]
+
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        emacs_version:
+          # - 28.2
+          # - 29.4
+          - 30.1
+          - snapshot
+
+    steps:
+    - uses: purcell/setup-emacs@master
+      with:
+        version: ${{ matrix.emacs_version }}
+
+    - name: Pull polymode
+      uses: actions/checkout@v4
+
+    - name: Pull poly-markdown
+      uses: actions/checkout@v4
+      with:
+        repository: polymode/poly-markdown
+        path: polyrepos/poly-markdown
+
+    - name: Pull poly-org
+      uses: actions/checkout@v4
+      with:
+        repository: polymode/poly-org
+        path: polyrepos/poly-org
+
+    - name: Pull poly-noweb
+      uses: actions/checkout@v4
+      with:
+        repository: polymode/poly-noweb
+        path: polyrepos/poly-noweb
+
+    - name: Check Emacs version
+      run: emacs --version
+
+    - name: ELPA
+      run: make elpa
+
+    - name: Build
+      run: make build
+
+    - name: Test
+      run: make test
```

Comment:
The last time when I tested, the `make test` pull package from package registry and thus, not testing the local version of `polymode`. That's why I'm using the `test-local` here

Reply:
yes, it's confusing. i just stumbled on it myself. It's because package-initialize puts all the directories into the load path. So we need an explicit overwrite of the path. 

`test-local` is what's needed here actually. Will need to simplify this setup at some point ..